clustermesh/clustermesh_test.go

```diff
@@ -0,0 +1,98 @@
+package clustermesh
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRemoveFromClustermeshConfig(t *testing.T) {
+	uu := map[string]struct {
+		vv      map[string]any
+		cluster string
+		err     error
+		e       map[string]any
+	}{
+		"missing": {
+			cluster: "test1",
+			err:     errors.New(`active cluster is not currently connected to cluster "test1"`),
```

Comment:
Why is this error different from the empty case below?

Reply:
Actually both in the `missing` and the `empty` cases the config is empty.

Reply:
No sure what you are getting at? Not quite, one use case is nil the other empty. This is testing the returns of the copy.

Reply:
Never mind, it was just a nitpick. My point is that since in both cases the `clusters` map is empty (whether it is nil or not is only an implementation detail IMHO) I would expect the same error.

Reply:
Got it! Tx Marco. No quite an implementation detail but a potential bug ie copy can return found=true and the copy=nil, hence the differentiation in the test and the fix in the original impl. Does this make sense?

Reply:
Yeah, the fix makes perfectly sense to me. I would just expect in both cases to immediately return with the same error, but it's really just a nitpick (related: https://github.com/cilium/cilium-cli/pull/1739#discussion_r1236503178)